### PR TITLE
changed input validation for login method

### PR DESCRIPTION
**Pull request by xHoudek**

### CODE REVIEW

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -114,12 +114,18 @@ const deleteUser = async (req, res, next) => {
...
-  if (!contype || contype.indexOf('application/json') !== 0) res.sendStatus(415);
-  else if (!inputValidation.hasAllAttrs(req.body, attrs)) res.status(400).send("Please provide required input attributes");
-  else if (inputValidation.includesNullorEmpty(req.body)) res.status(400).send("Please provide required input values");
-  else {
+  let success = true;
+  try {
+    //check content type
+    if(!inputValidation.checkContentType(req, res)) throw new ContentTypeError();
+    //check null values
+    errList = inputValidation.getNullorEmpty(res, req.body);
```
Comment (dotte-osu) [inline src/controllers/usersController.js:129]: Don't we need to check the length of the errList and throw an error?

Comment (dotte-osu) [reply src/controllers/usersController.js:129]: Also, don't we need to do the attributes check?

Comment (asching7108) [inline src/controllers/usersController.js:129]: Are we missing error throwing here? The errList is not used

**Review by dotte-osu — CHANGES_REQUESTED** [review]
Just a couple of things to verify before merging!

File: src/controllers/usersController.js (hunk 1 above)
Comment (xHoudek) [reply src/controllers/usersController.js:129]: Both updated, thanks!

Comment (xHoudek) [reply src/controllers/usersController.js:129]: Updated, thanks!

File: src/controllers/usersController.js [hunk 2]
```diff
@@ -68,9 +68,13 @@ const patchUser = async (req, res, next) => {
...
   try {
     //check content type
-    if(!inputValidation.checkContentType(req, res)) throw new ContentTypeError();
+    if (!inputValidation.checkContentType(req, res)) throw new ContentTypeError();
+    // all required attributes are provided
+    let errList = inputValidation.getMissingAttrs(res, req.body, ["firstName", "lastName", "email", "password", "zipCode"]);
+    if (errList.length != 0) throw new PropRequiredError(errList);
     //check null values
     errList = inputValidation.getNullorEmpty(res, req.body, ["address", "distancePreference"]);
+    if (errList.length != 0) throw new PropRequiredError(errList);
```
Comment (dotte-osu) [inline src/controllers/usersController.js]: I think this one should be `PropNullorEmptyError`

File: src/controllers/usersController.js [hunk 3]
```diff
@@ -114,12 +118,22 @@ const deleteUser = async (req, res, next) => {
...
+  let success = true;
+  try {
+    //check content type
+    if (!inputValidation.checkContentType(req, res)) throw new ContentTypeError();
+    // all required attributes are provided
+    let errList = inputValidation.getMissingAttrs(res, req.body, ["email", "password"]);
+    if (errList.length != 0) throw new PropRequiredError(errList);
+    //check null values
+    errList = inputValidation.getNullorEmpty(res, req.body);
+    if (errList.length != 0) throw new PropRequiredError(errList);
```
Comment (dotte-osu) [inline src/controllers/usersController.js]: I think this one is also `PropNullorEmptyError`...?

**Review by dotte-osu — APPROVED** [review]

**Pull request merged**